### DISCUSSION TITLE
Differentiating through quadratic program

### DISCUSSION THREAD

**Discussion by jcallaham** [casadi/casadi#3229] [Ideas]
I've been working on implementing AD for QP solvers in CasADi based on the [OptNet](https://arxiv.org/abs/1703.00443) paper .  For instance, this could be used to do differentiable multibody simulations using an LCP-type integrator.

If I get this working and put together a PR, would there be broader interest in adding this? I saw in the contribution guidelines that unsolicited PRs are likely to be ignored, so just wanted to check.

Thanks!
Jared

**Comment by jgillis** [comment]
@jaeandersson needs to answer this.. I would think that, since all the machinery is already in NLP, it would be rather easy carry this over to conic? Probably the hardest part is to avoid duplicating the code in a clean way..

For sure this is a desirable feature to have

**Comment by jaeandersson** [comment]
Yes, differentiating QPs would be great to have. And as @jgillis says, easy to implement since it's about generalizing the existing implementation available for NLP solvers. Unfortunately, don't have any current projects justifying spending time on this so I won't be able to work on it or review any PRs. Feel free to develop and maintain the feature in your own fork of CasADi though.
